File: 0121-best-time-to-buy-and-sell-stock/solution.go
func maxProfit(prices []int) int {
    low := prices[0]
    maxProfit := 0

    for _, price := range prices {
        if price < low {
            low = price
        }

        if price-low > maxProfit {
            maxProfit = price-low
        }
    }


    return maxProfit
}

File: 0121-best-time-to-buy-and-sell-stock/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int cheaper = prices[0];
        int profit = 0;
        
        for (int i : prices) {
            if (i < cheaper) {
                cheaper = i;
            }

            if (i  - cheaper > profit) {
                profit = i - cheaper;
            }
        }

        return profit;
    }
}
